test_fatoptimizer.py

```diff
@@ -3206,6 +3206,70 @@ def test_str_encode(self):
         self.check_dont_optimize(r'"ab\xff".encode("big5")')
         self.check_dont_optimize(r'"ab\xff".encode("ascii", "backslashreplace")')
 
+    def test_str_lower(self):
+        self.check_optimize(r'"ABC".lower()', '"abc"')
+
+    def test_str_upper(self):
+        self.check_optimize(r'"ABC".upper()', '"ABC"')
+
+    def test_str_capitalize(self):
+        self.check_optimize(r'"abc".capitalize()', '"Abc"')
+
+    def test_str_swapcase(self):
+        self.check_optimize(r'"aBc".swapcase()', '"AbC"')
+
+    def test_str_casefold(self):
+        self.check_optimize(r'"ABC".casefold()', '"abc"')
+        self.check_optimize(r'"ß".casefold()', '"ss"')
+
+    def test_str_isalpha(self):
+        self.check_optimize(r'"".isalpha()', 'False')
+        self.check_optimize(r'"123".isalpha()', 'False')
+        self.check_optimize(r'"abc123".isalpha()', 'False')
+        self.check_optimize(r'"abcß".isalpha()', 'True')
+
+    def test_str_isalnum(self):
+        self.check_optimize(r'"".isalnum()', 'False')
+        self.check_optimize(r'"abc123".isalnum()', 'True')
+        self.check_optimize(r'"abc123".isalnum()', 'True')
+        self.check_optimize(r'"abc123ß".isalnum()', 'True')
+
+    def test_str_isdecimal(self):
+        self.check_optimize(r'"".isdecimal()', 'False')
+        self.check_optimize(r'"abc".isdecimal()', 'False')
+        self.check_optimize(r'"\u00B2".isdecimal()', 'False')
+        self.check_optimize(r'"\u00BD".isdecimal()', 'False')
+        self.check_optimize(r'"1.0".isdecimal()', 'False')
+        self.check_optimize(r'"1".isdecimal()', 'True')
+
+    def test_str_isdigit(self):
+        self.check_optimize(r'"".isdigit()', 'False')
+        self.check_optimize(r'"abc".isdigit()', 'False')
+        self.check_optimize(r'"\u00B2".isdigit()', 'True')
+        self.check_optimize(r'"\u00BD".isdigit()', 'False')
+        self.check_optimize(r'"1.0".isdigit()', 'False')
+        self.check_optimize(r'"1".isdigit()', 'True')
+
+    def test_str_islower(self):
+        self.check_optimize(r'"".islower()', 'False')
+        self.check_optimize(r'"ABC".islower()', 'False')
+        self.check_optimize(r'"aBc".islower()', 'False')
+        self.check_optimize(r'"abc".islower()', 'True')
+
+    def test_str_isnumeric(self):
+        self.check_optimize(r'"".isnumeric()', 'False')
+        self.check_optimize(r'"abc".isnumeric()', 'False')
+        self.check_optimize(r'"\u00B2".isnumeric()', 'True')
+        self.check_optimize(r'"\u00BD".isnumeric()', 'True')
+        self.check_optimize(r'"1.0".isnumeric()', 'False')
+        self.check_optimize(r'"1".isnumeric()', 'True')
+
+    def test_str_isupper(self):
+        self.check_optimize(r'"".isupper()', 'False')
+        self.check_optimize(r'"ABC".isupper()', 'True')
+        self.check_optimize(r'"aBc".isupper()', 'False')
+        self.check_optimize(r'"abc".isupper()', 'False')
```

Comment:
Same here: fold is_xxx() into a single test function and restrict to one test per method.